openapi3filter/validate_request.go

```diff
@@ -76,10 +93,19 @@ func ValidateRequest(c context.Context, input *RequestValidationInput) error {
 		security = &route.Swagger.Security
 	}
 	if security != nil {
-		if err := ValidateSecurityRequirements(c, input, *security); err != nil {
+		if err = ValidateSecurityRequirements(c, input, *security); err != nil && !options.MultiError {
 			return err
 		}
+
+		if err != nil {
+			me = append(me, err)
+		}
 	}
+
+	if len(me) > 0 {
```

Comment:
How about just `return me`?

Reply:
This actually won't work in the go type system. If we were to return `me`, future `if err != nil` would always evaluate to true. See this [playground](https://play.golang.org/p/lBLb7FdKFsJ) as a brief illustration.

The reason is because in the go type system, [interfaces hold both underlying type and value](https://codefibers.pl/blog/golang-why-nil-is-not-always-nil). So in this case we would need to return nil explicitly since the function returns an error interface. Always returning `me` might have a nil value, but would always have a non-nil type meaning it would never satisfy equality (`==` ) with `nil`